lgc/disassembler/PalMetadataRegs.cpp

```diff
@@ -0,0 +1,405 @@
+    {0x2E2A, "COMPUTE_SHADER_CHKSUM"},
+    {0x2E40, "COMPUTE_USER_DATA_0"},
+    {0x2E41, "COMPUTE_USER_DATA_1"},
+    {0x2E42, "COMPUTE_USER_DATA_2"},
+    {0x2E43, "COMPUTE_USER_DATA_3"},
+    {0x2E44, "COMPUTE_USER_DATA_4"},
+    {0x2E45, "COMPUTE_USER_DATA_5"},
+    {0x2E46, "COMPUTE_USER_DATA_6"},
+    {0x2E47, "COMPUTE_USER_DATA_7"},
+    {0x2E48, "COMPUTE_USER_DATA_8"},
+    {0x2E49, "COMPUTE_USER_DATA_9"},
+    {0x2E4A, "COMPUTE_USER_DATA_10"},
+    {0x2E4B, "COMPUTE_USER_DATA_11"},
+    {0x2E4C, "COMPUTE_USER_DATA_12"},
+    {0x2E4D, "COMPUTE_USER_DATA_13"},
+    {0x2E4E, "COMPUTE_USER_DATA_14"},
+    {0x2E4F, "COMPUTE_USER_DATA_15"},
+    {0xA08F, "CB_SHADER_MASK"},
+    {0xA191, "SPI_PS_INPUT_CNTL_0"},
+    {0xA192, "SPI_PS_INPUT_CNTL_1"},
+    {0xA193, "SPI_PS_INPUT_CNTL_2"},
+    {0xA194, "SPI_PS_INPUT_CNTL_3"},
+    {0xA195, "SPI_PS_INPUT_CNTL_4"},
+    {0xA196, "SPI_PS_INPUT_CNTL_5"},
+    {0xA197, "SPI_PS_INPUT_CNTL_6"},
+    {0xA198, "SPI_PS_INPUT_CNTL_7"},
+    {0xA199, "SPI_PS_INPUT_CNTL_8"},
+    {0xA19A, "SPI_PS_INPUT_CNTL_9"},
+    {0xA19B, "SPI_PS_INPUT_CNTL_10"},
+    {0xA19C, "SPI_PS_INPUT_CNTL_11"},
+    {0xA19D, "SPI_PS_INPUT_CNTL_12"},
+    {0xA19E, "SPI_PS_INPUT_CNTL_13"},
+    {0xA19F, "SPI_PS_INPUT_CNTL_14"},
+    {0xA1A0, "SPI_PS_INPUT_CNTL_15"},
+    {0xA1A1, "SPI_PS_INPUT_CNTL_16"},
+    {0xA1A2, "SPI_PS_INPUT_CNTL_17"},
+    {0xA1A3, "SPI_PS_INPUT_CNTL_18"},
+    {0xA1A4, "SPI_PS_INPUT_CNTL_19"},
+    {0xA1A5, "SPI_PS_INPUT_CNTL_20"},
+    {0xA1A6, "SPI_PS_INPUT_CNTL_21"},
+    {0xA1A7, "SPI_PS_INPUT_CNTL_22"},
+    {0xA1A8, "SPI_PS_INPUT_CNTL_23"},
+    {0xA1A9, "SPI_PS_INPUT_CNTL_24"},
+    {0xA1AA, "SPI_PS_INPUT_CNTL_25"},
+    {0xA1AB, "SPI_PS_INPUT_CNTL_26"},
+    {0xA1AC, "SPI_PS_INPUT_CNTL_27"},
+    {0xA1AD, "SPI_PS_INPUT_CNTL_28"},
+    {0xA1AE, "SPI_PS_INPUT_CNTL_29"},
+    {0xA1AF, "SPI_PS_INPUT_CNTL_30"},
+    {0xA1B0, "SPI_PS_INPUT_CNTL_31"},
+    {0xA1B1, "SPI_VS_OUT_CONFIG"},
+    {0xA1B3, "SPI_PS_INPUT_ENA"},
+    {0xA1B4, "SPI_PS_INPUT_ADDR"},
+    {0xA1B5, "SPI_INTERP_CONTROL_0"},
+    {0xA1B6, "SPI_PS_IN_CONTROL"},
+    {0xA1B8, "SPI_BARYC_CNTL"},
+    {0xA1BA, "SPI_TMPRING_SIZE"},
+    {0xA1C2, "SPI_SHADER_IDX_FORMAT"},
+    {0xA1C3, "SPI_SHADER_POS_FORMAT"},
+    {0xA1C4, "SPI_SHADER_Z_FORMAT"},
+    {0xA1C5, "SPI_SHADER_COL_FORMAT"},
+    {0xA1FF, "GE_MAX_OUTPUT_PER_SUBGROUP"},
+    {0xA203, "DB_SHADER_CONTROL"},
+    {0xA204, "PA_CL_CLIP_CNTL"},
+    {0xA206, "PA_CL_VTE_CNTL"},
+    {0xA207, "PA_CL_VS_OUT_CNTL"},
+    {0xA210, "PA_STEREO_CNTL"},
+    {0xA286, "VGT_HOS_MAX_TESS_LEVEL"},
+    {0xA287, "VGT_HOS_MIN_TESS_LEVEL"},
+    {0xA290, "VGT_GS_MODE"},
+    {0xA291, "VGT_GS_ONCHIP_CNTL"},
+    {0xA293, "PA_SC_MODE_CNTL_1"},
+    {0xA295, "VGT_GS_PER_ES"},
+    {0xA296, "VGT_ES_PER_GS"},
+    {0xA297, "VGT_GS_PER_VS"},
+    {0xA298, "VGT_GSVS_RING_OFFSET_1"},
+    {0xA299, "VGT_GSVS_RING_OFFSET_2"},
+    {0xA29A, "VGT_GSVS_RING_OFFSET_3"},
+    {0xA29B, "VGT_GS_OUT_PRIM_TYPE"},
```

Comment:
I think we do not need a robust solution for that here, because:

1. This is only meant to be a helpful human-readable indication of what the register is. The table is not used for name->number mapping, only number->name mapping. The way the registers are output in the disassembly means that the mnemonic name gets ignored if you try and reassemble.

2. This shouldn't be here at all; we should instead be using the same functionality in the PAL metadata object in the AMDGPU back-end. However, as that is in target-specific LLVM code, there is no way to get to it from here. I tried to create a way to get to it, but it got stuck in phab review three years ago: https://reviews.llvm.org/D52822 . I need to go back and resurrect that, so we can remove this table here.

If I can do that, and update the table there, then it is worth thinking about adding a mechanism for small changes between hardware generations.

Reply:
Thanks for clarifying this. I'm OK now.